src/SourceCode.Clay.AspNetCore.Middleware.Correlation/CorrelationContext.cs

```diff
@@ -12,9 +12,8 @@ internal CorrelationContext(string correlationId, string header)
         {
             if (string.IsNullOrWhiteSpace(correlationId))
                 throw new ArgumentException(ArgumentNullOrWhitespace, nameof(correlationId));
-
-            if (string.IsNullOrWhiteSpace(header))
-                throw new ArgumentException(ArgumentNullOrWhitespace, nameof(header));
+            if (header != null && string.IsNullOrWhiteSpace(header))
+                throw new ArgumentOutOfRangeException(ArgumentNullOrWhitespace, nameof(CorrelationId));

```

Comment:
Undo this change please

Reply:
What should the value be if no header is found?

Reply:
Ignore; I have a plan